tests/test-inference.js

```diff
@@ -43,19 +43,44 @@ function runDiscreteSamplingTest(test, code, expectedHist, numSamples, tolerance
   }
 }
 
-function runDistributionTest(test, code, expectedHist, tolerance) {
+function getHist(erp) {
   var hist = {};
+  erp.support().forEach(
+      function(value) {
+        hist[value] = Math.exp(erp.score([], value));
+      });
+  return util.normalizeHist(hist);
+}
+
+function runDistributionTest(test, testData) {
   var k = function(s, erp) {
-    erp.support().forEach(
-        function(value) {
-          hist[value] = Math.exp(erp.score([], value));
-        });
-    var normHist = util.normalizeHist(hist);
-    test.ok(util.histsApproximatelyEqual(normHist, expectedHist, tolerance));
+    var normHist = getHist(erp);
+    var expected = testData.expected;
+    test.ok(util.histsApproximatelyEqual(normHist, expected.hist, expected.tolerance));
+    test.done();
+  };
+
+  webppl.run(testData.code, k);
+}
+
+function runDistributionStatisticsTest(test, testData) {
+  var k = function(s, erp) {
+    var normHist = getHist(erp);
+    var testMean = util.expectation(normHist);
+    var testStd = util.std(normHist);
+    var expected = testData.expected;
+    var allOk = true;
+    if (testMean - testStd < expected.mean - expected.std ||
+        testMean + testStd > expected.mean + expected.std) {
```

Comment:
I'm confused by this condition. I expected something like

    testMean < expected.mean - expected.std

or

    testMean < expected.mean - tolerance

It seems that you are comparing (actual mean - actual std) to (expected mean - expected std). Shouldn't we expect these quantities to be approximately equal, and so the test to fail half the time, even if it has the correct mean and std?